spec/system/user_authentication_spec.rb

```diff
@@ -7,11 +7,23 @@
   scenario 'when signing up' do
     visit '/'
     click_on 'Sign Up'
+    fill_in 'Username', with: 'Mentessi'
     fill_in 'Email', with: ''
     fill_in 'Password', with: 'password'
     fill_in 'Password confirmation', with: 'password'
     click_button 'Sign up'
     expect(page).to have_content "Email can't be blank"
+
+    visit '/'
+    click_on 'Sign Up'
+    fill_in 'Username', with: ''
+    fill_in 'Email', with: 'michael@bluespot.io'
```

Comment:
You should be able to test all the missing required fields at once for a more performant spec (i.e. no need for an entirely new visit to the page just to check the Username field can't be blank - just submit the form with email *and* username blank and check the messages).